### PR TITLE
Bridge ResteasyConfiguration and MP Config for standalone

### LINKED ISSUE #9671: GZip filter doesn't work
**Describe the bug**
I followed the steps in https://quarkus.io/guides/rest-json#gzip-support but the filter doesn't seem to be activated.

**Expected behavior**
Content will be compressed as GZIP 

**Actual behavior**
No GZIP compression

**To Reproduce**
1. Create any endpoint with code.quarkus.io
2. Add `quarkus.resteasy.gzip.enabled=true` to application.properties
3. Run `http localhost:8080/hello Content-Encoding:gzip Accept-Encoding:gzip`

**Configuration**
```properties
quarkus.resteasy.gzip.enabled=true
```
**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Fedora 32
 - Output of `java -version`: OpenJDK 11
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.5.0.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Maven 3.6.3 

**Additional context**

It looks like the `Content-Encoding` and `Accept-Encoding` headers are not propagated to the internal Vert.x Request, thus making `org.jboss.resteasy.plugins.interceptors.GZIPEncodingInterceptor` ignore the request to GZIP the output

### PR DESCRIPTION
**Pull request by patriot1burke**
Resteasy has an SPI for internal configuration for many different resteasy settings and components.  Resteasy + servlet got its config from servlet parameters in web.xml.  Resteasy standalone was never implemented to get config from anywhere.

### CODE REVIEW

File: extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/ResteasyConfigurationMPConfig.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * Some resteasy components use this class for configuration. This bridges MP Config to ResteasyConfiguration
+ *
+ */
+public class ResteasyConfigurationMPConfig implements ResteasyConfiguration {
+    @Override
+    public String getParameter(String name) {
+        Config config = ConfigProvider.getConfig();
+        if (config == null)
+            return null;
+        Optional<String> val = config.getOptionalValue(name, String.class);
```
Comment (gastaldi) [inline extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/ResteasyConfigurationMPConfig.java:22]:
```suggestion
        return config.getOptionalValue(name, String.class).orElse(null);;
```

File: integration-tests/resteasy-jackson/src/main/resources/application.properties [hunk 2]
```diff
@@ -1,2 +1,3 @@
 message=Production
-quarkus.smallrye-openapi.store-schema-directory=target/generated
\ No newline at end of file
+quarkus.smallrye-openapi.store-schema-directory=target/generated
+resteasy.gzip.max.input=5000
```
Comment (gsmet) [inline integration-tests/resteasy-jackson/src/main/resources/application.properties:3]: I'm not sure we want to do it this way. My guess is that we would want some proper Quarkus config declared via a proper config class for that.

@stuartwdouglas WDYT?

**Review by gsmet — DISMISSED** [review]
Marking it as request changes to get Stuart's opinion.

File: integration-tests/resteasy-jackson/src/main/resources/application.properties (hunk 2 above)
Comment (patriot1burke) [reply integration-tests/resteasy-jackson/src/main/resources/application.properties:3]: See initial comment .  Resteasy runtime uses the ResteasyConfiguration SPI.

Comment (patriot1burke) [reply integration-tests/resteasy-jackson/src/main/resources/application.properties:3]: @gsmet Also, that was just a test example.  Could have been any value for the test.

**Comment by patriot1burke** [comment]
@gsmet Bump.  See my comment above.  Resteasy uses the ResteasyConfiguration internally in a bunch of places and there is no way to configure those components within standalone.

File: integration-tests/resteasy-jackson/src/main/resources/application.properties (hunk 2 above)
Comment (stuartwdouglas) [reply integration-tests/resteasy-jackson/src/main/resources/application.properties:3]: We do expose this under quarkus.resteasy.gzip.max-input

**Review by stuartwdouglas — APPROVED** [review]

**Comment by patriot1burke** [comment]
@gsmet  Need you to resolve your request.

**Pull request merged**